### PR TITLE
Venues

### PR DESCRIPTION
**Pull request by Sam-Archie**
## Description

Getting venues working in a basic manner plus some boiler plate for search functions and pagination once we have the BE work complete.

Here is what it looks like:
Screenshot 2022-11-22 at 15 11 50
Screenshot 2022-11-22 at 15 11 54
Screenshot 2022-11-22 at 15 13 49
Screenshot 2022-11-22 at 15 13 52
Screenshot 2022-11-22 at 15 14 00

### CODE REVIEW

File: @types/index.d.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ type RegisterUserMutationData = {
   email: string;
   first_name: string;
   last_name: string;
+  z;
```
Comment (ruarim) [inline @types/index.d.ts]: whats `z`?

File: @types/index.d.ts [hunk 2]
```diff
@@ -35,3 +36,47 @@ type LoginResponse = {
...
+  opening_time: string;
+  venue_type: "Pub" | "Bar" | "Restaurant";
+  address: Address;
+  attributes: Attribute[];
+  beverages: Beverage[];
+};
+
+type VenueResponse = {
+  data: {
+    data: Venue[];
```
Comment (ruarim) [inline @types/index.d.ts:53]: how come this is nested like this?

File: pages/venues/index.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+          <p className="mx-auto mt-3 max-w-2xl text-xl text-gray-500 sm:mt-4">
+            Search through to find the perfect spot.
+          </p>
+        </div>
+        <div className="pt-12">
+          <SearchInput />
+        </div>
+        <div className="mx-auto pt-6 grid max-w-lg gap-5 lg:max-w-none lg:grid-cols-3">
+          {venueData &&
+            venueData.data.data.map((venue: Venue) => {
```
Comment (ruarim) [inline pages/venues/index.tsx:28]: guess this is from the nested type?

**Review by ruarim — APPROVED** [review]
couple questions, looks great otherwise

File: @types/index.d.ts (hunk 2 above)
Comment (Sam-Archie) [reply @types/index.d.ts:53]: That is the shape of the data when it comes out of the api.

File: @types/index.d.ts (hunk 1 above)
Comment (Sam-Archie) [reply @types/index.d.ts]: typo. good catch

File: pages/venues/index.tsx (hunk 3 above)
Comment (Sam-Archie) [reply pages/venues/index.tsx:28]: yes indeed.

**Pull request merged**